deflate_slow.c

```diff
@@ -106,7 +106,17 @@ ZLIB_INTERNAL block_state deflate_slow(deflate_state *s, int flush) {
             s->match_length = MIN_MATCH-1;
             s->strstart++;
 #else
-            {
+            if (s->lookahead < MIN_MATCH) {
+                s->prev_length -= 2;
+                do {
+                    if (++s->strstart <= max_insert) {
+                        functable.insert_string(s, s->strstart, 1);
+                    }
+                } while (--s->prev_length != 0);
```

Comment:
Calling insert_string inside do-while loop is very inefficient, so should be avoided. I'm not sure this is the best way to fix the issue. 
Shouldn't we calculate minimum of ```s->prev_length``` and ```max_insert - s->strstart```?
The problem in the original code was that it used ```unsigned int``` in the substraction when it should have used ```int```.

Reply:
You are correct.  See updated patch 0352a24.